contracts/reverseRegistrar/SignatureReverseResolver.sol

```diff
@@ -0,0 +1,264 @@
+pragma solidity >=0.8.4;
+
+import "../registry/ENS.sol";
+import "./ISignatureReverseResolver.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../root/Controllable.sol";
+import "../utils/LowLevelCallUtils.sol";
+
+error InvalidSignature();
+error SignatureOutOfDate();
+error Unauthorised();
+
+contract SignatureReverseResolver is Ownable, ISignatureReverseResolver {
+    using ECDSA for bytes32;
+    mapping(bytes32 => uint256) public lastUpdated;
+    mapping(uint64 => mapping(bytes32 => mapping(string => string))) versionable_texts;
+    mapping(uint64 => mapping(bytes32 => string)) versionable_names;
+    mapping(bytes32 => uint64) internal recordVersions;
+
+    bytes32 public immutable parentNode;
+    uint256 public immutable coinType;
+
+    /*
+     * @dev Constructor
+     * @param parentNode The namespace to set.
+     * @param _coinType The cointype converted from the chainId of the chain this contract is deployed to.
+     */
+    constructor(bytes32 _parentNode, uint256 _coinType) {
+        parentNode = _parentNode;
+        coinType = _coinType;
+    }
+
+    modifier authorised(address addr) {
+        isAuthorised(addr);
+        _;
+    }
+
+    modifier authorisedSignature(
+        bytes32 hash,
+        address addr,
+        uint256 inceptionDate,
+        bytes memory signature
+    ) {
+        isAuthorisedWithSignature(hash, addr, inceptionDate, signature);
+        _;
+    }
+
+    function getLastUpdated(
```

Comment:
We don't use `get` as a prefix elsewhere, we probably shouldn't here.

Reply:
Looks not in use so removed